src/nanoemoji/glue_together.py

```diff
@@ -58,13 +56,45 @@ def _collect_glyphs(paint):
     target["COLR"] = donor["COLR"]
 
 
+def _svg_glyphs(font: ttLib.TTFont) -> Iterable[Tuple[int, str]]:
+    for _, min_gid, max_gid in font["SVG "].docList:
+        for gid in range(min_gid, max_gid + 1):
+            yield gid, font.getGlyphName(gid)
+
+
+def _copy_svg(target: ttLib.TTFont, donor: ttLib.TTFont):
+    # SVG is exciting because nanoemoji likes to restructure glyph order
+    # To keep things simple, let's build a new glyph order that keeps all the svg font gids stable
+    target_glyph_order = list(target.getGlyphOrder())
+    svg_glyphs = {gn for _, gn in _svg_glyphs(donor)}
+    non_svg_target_glyphs = [gn for gn in target_glyph_order if gn not in svg_glyphs]
+    new_glyph_order = []
+
+    for svg_gid, svg_glyph_name in _svg_glyphs(donor):
+        # we want gid to remain stable so copy non-svg glyphs until that will be true
+        while len(new_glyph_order) < svg_gid:
+            new_glyph_order.append(non_svg_target_glyphs.pop(0))
```

Comment:
I think this while loop is never actually triggered because all the svg glyphs are placed in a contiguous block of gids at the end (see `nanoemoji.svg._ensure_groups_grouped_in_glyph_order`). Maybe an assert bould be just fine.
This glue_together module is for internal use only and the "donor" font is one that nanoemoji itself has built.

Reply:
oh I see what's going on here. Even if we know all the svg color glyphs are in a contiguous block in the donor's glyphOrder (which nanoemoji itself has built), we don't control the target's glyphOrder which can be anything; and at the same time, we don't want to have to remap the `id="glyphX"` attributes in the OT-SVG documents. So when glueing together the two fonts make sure the glyph ids of for the SVG glyphs to stay the same as in the donor font and reshuffle the target's glyphs around accordingly.